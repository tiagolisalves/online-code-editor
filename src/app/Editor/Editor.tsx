import React, { useEffect } from 'react'
// import hljs from 'highlight.js/lib/languages/java'
import hljs from 'highlight.js'
import java from 'highlight.js/lib/languages/java'
hljs.registerLanguage('java', java)

function Editor(props: { fileContent: string; onEdit: any }) {
    let editable: HTMLElement | null
    useEffect(() => {
        if (editable) {
            hljs.highlightBlock(editable)
        }
    })

    const onEdit = (event: React.FormEvent<HTMLDivElement>) => {
        let fileContentParam = (event.currentTarget as any).innerText
        props.onEdit(fileContentParam)
    }

    return (
        <div contentEditable="true" className="editor-area" onInput={onEdit} suppressContentEditableWarning={true}>
            <pre>
                <code
                    ref={(editableRef) => {
                        editable = editableRef
                    }}
                >
                    {props.fileContent}
                </code>
            </pre>
        </div>
    )
}

export { Editor }
