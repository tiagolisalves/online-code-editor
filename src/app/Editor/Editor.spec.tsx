import { create } from 'react-test-renderer'
import React from 'react'
import { Editor } from './Editor'

test('It should get filetree from parent', () => {
    const comp = create(<Editor fileContent={fixture().content} onEdit={() => {}}></Editor>)
    expect(comp.toJSON()).toMatchSnapshot()
})

function fixture() {
    return {
        id: 4,
        name: 'Main.java',
        content:
            '/*\r\n * Copyright 2012 Veezoo, Inc.\r\n *\r\n *    Licensed under the Apache License, Version 2.0 (the "License");\r\n *    you may not use this file except in compliance with the License.\r\n *    You may obtain a copy of the License at\r\n *\r\n *        http://www.apache.org/licenses/LICENSE-2.0\r\n */\r\n\r\npackage com.veezoo.main;\r\n\r\nimport java.io.IOException;\r\nimport java.util.Arrays;\r\nimport java.util.Collections;\r\n\r\nimport javax.annotation.Nullable;\r\nimport javax.inject.Singleton;\r\n\r\nimport org.slf4j.Logger;\r\nimport org.slf4j.LoggerFactory;\r\n\r\n/**\r\n *\r\n * A default implementation of Main server as required by\r\n * {@link MainPropertyFactory}.\r\n *\r\n */\r\n@Singleton\r\npublic class Main implements AbstractMain {\r\n    private static final String TEST = "test";\r\n    private static final Logger logger = LoggerFactory.getLogger(Main.class);\r\n    private static final MainPropertyFactory configInstance = com.veezoo.Main.MainPropertyFactory\r\n            .getInstance();\r\n    private static final int TIME_TO_WAIT_FOR_GREETING = 30000;\r\n\r\n    public static void main(String[] args) {\r\n        // do nothing\r\n    }\r\n\r\n}',
    }
}
