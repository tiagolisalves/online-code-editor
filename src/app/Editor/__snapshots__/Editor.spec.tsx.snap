// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should get filetree from parent 1`] = `
<div
  className="editor-area"
  contentEditable="true"
  onInput={[Function]}
  suppressContentEditableWarning={true}
>
  <pre>
    <code>
      /*
 * Copyright 2012 Veezoo, Inc.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 */

package com.veezoo.main;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collections;

import javax.annotation.Nullable;
import javax.inject.Singleton;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 *
 * A default implementation of Main server as required by
 * {@link MainPropertyFactory}.
 *
 */
@Singleton
public class Main implements AbstractMain {
    private static final String TEST = "test";
    private static final Logger logger = LoggerFactory.getLogger(Main.class);
    private static final MainPropertyFactory configInstance = com.veezoo.Main.MainPropertyFactory
            .getInstance();
    private static final int TIME_TO_WAIT_FOR_GREETING = 30000;

    public static void main(String[] args) {
        // do nothing
    }

}
    </code>
  </pre>
</div>
`;
